engine/src/execution_graph/logic_controllers/BatchProcessing.h

```diff
@@ -307,7 +307,7 @@ class DataSourceSequence {
 	}
 
 	bool has_next() {
-		return (is_empty_data_source && batch_index < 1) || (is_gdf_parser && batch_index.load() < n_batches) || (cur_file_index < n_files);
+		return (is_empty_data_source && batch_index.load() < 1) || (is_gdf_parser && batch_index.load() < n_batches) || (cur_file_index < n_files);
```

Comment:
why did you add the load here instead of comparing to the atomic itself. this should not matter when comparing to a literal.

Reply:
https://stackoverflow.com/questions/18850752/must-i-call-atomic-load-store-explicitly

Reply:
or any of these actually

Reply:
Using load or not does not matter. It does the same underneath, as your stack overflow post suggests. But using load explicitly indicates that its an atomic